src/services/goToDefinition.ts

```diff
@@ -32,6 +32,13 @@ namespace ts.GoToDefinition {
             return functionDeclaration ? [createDefinitionFromSignatureDeclaration(typeChecker, functionDeclaration)] : undefined;
         }
 
+        if (node.kind === SyntaxKind.CaseKeyword || node.kind === SyntaxKind.DefaultKeyword) {
+            const switchStatement = findAncestor(node.parent, n => isExportAssignment(n) ? "quit" : isSwitchStatement(n)) as SwitchStatement | undefined;

```

Comment:
This might work; however, here's how I might write it:

```ts
switch (node.kind) {
    case SyntaxKind.ReturnKeyword:
        // copy from above
        break;
    case SyntaxKind.DefaultKeyword:
        if (!isDefaultClause(node.parent)) {
            break;
        }
        // fallthrough
    case SyntaxKind.CaseKeyword:
        // current logic below
        break;
}
```

Reply:
Looks prettier